### PR TITLE
[FLINK-27097] Document custom validator implementations

### PR DESCRIPTION
**Pull request by SteNicholas**
Currently we allow users to specify custom validator implementations through the operator configuration. This can be very useful in production environments to implement custom validation logic that is specific to certain envs. How to custom validator implementations should have document to introduce.

**The brief change log**

- Adds the `validator.md` to document how to custom the `FlinkResourceValidator` implementation.

### CODE REVIEW

**Comment by SteNicholas** [comment]
@gyfora, PTAL.

File: docs/content/docs/operations/validator.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ```
+
+2. Create service definition file `org.apache.flink.kubernetes.operator.validation.FlinkResourceValidator` in `META-INF/services`.   With custom `FlinkResourceValidator` implementation, the service definition describes as follows:
+   ```text
+   org.apache.flink.kubernetes.operator.validation.CustomValidator
+   ```
+
+3. Use the Maven tool to package the project and upload the generated JAR to the `/opt/flink/plugins` directory of the mirror.
+
+4. Configure the flink-kubernetes-operator helm chart with the `FLINK_PLUGINS_DIR` environment variable:
```
Comment (Aitozi) [inline docs/content/docs/operations/validator.md:64]: The env have been defined in the helm chart, users seems do not have to do this.

File: docs/content/docs/operations/validator.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+       }
+   }
+    ```
+
+2. Create service definition file `org.apache.flink.kubernetes.operator.validation.FlinkResourceValidator` in `META-INF/services`.   With custom `FlinkResourceValidator` implementation, the service definition describes as follows:
+   ```text
+   org.apache.flink.kubernetes.operator.validation.CustomValidator
+   ```
+
+3. Use the Maven tool to package the project and upload the generated JAR to the `/opt/flink/plugins` directory of the mirror.
```
Comment (Aitozi) [inline docs/content/docs/operations/validator.md:62]: when I test last time, I found that the PluginManager use the `DirectoryBasedPluginFinder`. It seems only work when the plugins are put under a plugin directory as below:

```
 * plugins-root-folder/
 *            |------------validator1/ (folder of validator)
 *            |                |-custom-validator-1.jar
 *            |                |-...
```

So we should create a validator director before copy the custom validator jar to, for example: `/opt/flink/plugins/validator1/`

Comment (Aitozi) [reply docs/content/docs/operations/validator.md:62]: Can we directly add an example to tell user can `FROM` the `flink-kubernetes-operator` official image and `COPY` the plugin jars to the plugin dir. In this way, user can make their own operator image with custom validator plugin, Right?

File: docs/content/docs/operations/validator.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Custom `FlinkResourceValidator` implementation
+
+`FlinkResourceValidator`, a validator interface for resources of `FlinkDeployment` and `FlinkSessionJob`,  is pluggable component based on the [Plugins](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/filesystems/plugins) mechanism. During development, we can custom the implementation of `FlinkResourceValidator` and make sure to retain the service definition in `META-INF/services`.
```
Comment (wangyang0918) [inline docs/content/docs/operations/validator.md:29]:
```suggestion
`FlinkResourceValidator`, an interface for validating the resources of `FlinkDeployment` and `FlinkSessionJob`, is a pluggable component based on the [Plugins](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/filesystems/plugins) mechanism. 
```

File: docs/content/docs/operations/validator.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Custom `FlinkResourceValidator` implementation
+
+`FlinkResourceValidator`, a validator interface for resources of `FlinkDeployment` and `FlinkSessionJob`,  is pluggable component based on the [Plugins](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/filesystems/plugins) mechanism. During development, we can custom the implementation of `FlinkResourceValidator` and make sure to retain the service definition in `META-INF/services`. 
+To use a pluggable validator, copy the corresponding JAR file to a directory under plugins directory which is `/opt/flink/plugins` at default. The following steps demonstrate the custom validator implementation process and resources of `FlinkDeployment` and `FlinkSessionJob` are validated in custom way with the custom validator:
```
Comment (wangyang0918) [inline docs/content/docs/operations/validator.md:30]:
```suggestion
The following steps demonstrate how to develop and use a custom validator.
```

File: docs/content/docs/operations/validator.md (hunk 1 above)
Comment (wangyang0918) [reply docs/content/docs/operations/validator.md:64]: Yes. We already have the `FLINK_PLUGINS_DIR` env.

File: docs/content/docs/operations/validator.md (hunk 2 above)
Comment (wangyang0918) [reply docs/content/docs/operations/validator.md:62]: I think @Aitozi is right. The custom validator needs to be put in a dedicated directory.

Either users need to build a custom image, or use the init-container via `kustomize` to download the validator plugins.

File: docs/content/docs/operations/validator.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    import org.apache.flink.kubernetes.operator.crd.FlinkSessionJob;
+   
+    import java.util.Optional;
+   
+    /** Custom validator implementation of {@link FlinkResourceValidator}. */
+    public class CustomValidator implements FlinkResourceValidator {
+   
+        @Override
+        public Optional<String> validateDeployment(FlinkDeployment deployment) {
+            return Optional.empty();
```
Comment (gyfora) [inline docs/content/docs/operations/validator.md:46]: Can we demonstrate some actual custom validation here? A simple if check with an error would be enough

File: docs/content/docs/operations/validator.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Custom `FlinkResourceValidator` implementation
+
+`FlinkResourceValidator`, an interface for validating the resources of `FlinkDeployment` and `FlinkSessionJob`,  is a pluggable component based on the [Plugins](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/filesystems/plugins) mechanism. During development, we can custom the implementation of `FlinkResourceValidator` and make sure to retain the service definition in `META-INF/services`.
```
Comment (gyfora) [inline docs/content/docs/operations/validator.md:29]: typo: `we can custom the implementation` -> `we can customize the implementation`

**Review by wangyang0918 — APPROVED** [review]
+1 for merging.

**Pull request merged**